include/live_model.h

```diff
@@ -79,9 +80,10 @@ namespace reinforcement_learning {
      * Initialize the library and start the background threads used for
      * model managment and sending actions and outcomes to the online trainer
      * @param status  Optional field with detailed string description if there is an error
+     * @param binding_trace_callback Optional trace logger callback
      * @return int Return error code.  This will also be returned in the api_status object
      */
-    int init(api_status* status=nullptr);
+    int init(api_status* status=nullptr, trace_callback binding_trace_callback=nullptr);
```

Comment:
Instead of passing in a trace logger, please use the factory and config settings to create a logger.

Reply:
This is not a i_trace logger implementation, but a delegate of typedef void(callback*)(int, char*). See comments in live_model.h. Let's have one place to discuss.